File: Guests/Kotlin/sanity/TestFuncs.kt
package sanity;

class TestFuncs{
	companion object{

		@JvmStatic
		fun helloWorld() {
			println("Hello World - From Java");
		}

		@JvmStatic
		fun returnsAnError() {
			throw Exception("Returning an error");
		}

		@JvmStatic
		fun divIntegers(x: Int, y: Int): Float {
			if(y == 0)
				throw ArithmeticException("Divisor is 0");

			return x.toFloat().div(y.toFloat());
		}

		@JvmStatic
		fun joinStrings(arr: Array<String>): String{
			return arr.joinToString(",")
		}
	}
}

File: Guests/Scala/sanity/TestFuncs.scala
package sanity;

class TestFuncs; object TestFuncs{
	def helloWorld() = {
		System.out.println("Hello World - From Scala");
	}

	def returnsAnError() = {
		throw new Exception("Returning an error");
	}

	def divIntegers(x: Int, y: Int): Float = {
		if(y == 0)
			throw new ArithmeticException("Divisor is 0");

		return x.toFloat / y.toFloat;
	}

	def joinStrings(arr: Array[String]): String = {
		return arr.mkString(",")
	}
}